src/platform/plugins/shared/workflows_management/server/workflows_management/workflows_management_api.ts

```diff
@@ -200,18 +209,38 @@ export class WorkflowsManagementApi {
     return executeResponse.workflowExecutionId;
   }
 
-  public async testWorkflow(
-    workflowYaml: string,
-    inputs: Record<string, any>,
-    spaceId: string,
-    request: KibanaRequest
-  ): Promise<string> {
+  public async testWorkflow({
+    workflowId,
+    workflowYaml,
+    inputs,
+    spaceId,
+    request,
+  }: TestWorkflowParams): Promise<string> {
+    let resolvedYaml = workflowYaml;
+    let resolvedWorkflowId = workflowId;
+
+    if (workflowId && !workflowYaml) {
+      const existingWorkflow = await this.workflowsService.getWorkflow(workflowId, spaceId);
+      if (!existingWorkflow) {
+        throw new WorkflowNotFoundError(workflowId);
+      }
+      resolvedYaml = existingWorkflow.yaml;
+    }
+
+    if (!resolvedWorkflowId) {
+      resolvedWorkflowId = 'test-workflow';
```

Comment:
maybe to to create use the uuid prefixed with test-workflow instead of always 'test-workflow'?